libs/stats/odc/stats/_fc_percentiles.py

```diff
@@ -41,40 +41,38 @@ def measurements(self) -> Tuple[str, ...]:
     @staticmethod
     def _native_tr(xx):
         """
-        Loads in the data in the native projection. It performs the following:
-
-        1. Loads all the fc and WOfS bands
-        2. Set high slope terrain flag and nodata wofs to 0
-        3. Extracts the clear dry and clear wet flags from WOfS
-        4. Drops the WOfS band
-        5. Masks out all pixels that are not clear and dry to a nodata value of 255
-        6. Discards the clear dry flags
-        """
-
-        # Set terrain and nodata flag to zero
-        #
-        # Leaving the nodata flag causes any pixel that has a nodata value for
-        # one or more wofs scenes on a given day to be set to nodata for that day
-        #
-        # This a particular issue because large areas on the edges of scenes 
-        # are nodata. This causes any pixel that overlaps with the nodata edge region
-        # of another scene to always be set to nodata
-
-        water = da.bitwise_and(xx["water"], 0b1110_1110)
-        xx = xx.drop_vars(["water"])
+        Loads data in its native projection. It performs the following:
 
+        1. Load all the fc and WOfS bands
+        2. Set the high terrain slope flag to 0
+        3. Find bad WOfS pixels ignoring the nodata flags
+        4. Extracts the clear dry and clear wet WoFS pixels
+        5. Drops the WOfS band
+        """
+        
+        water = xx.water& 0b1110_1111
+        xx["bad"] = (water & 0b0111_1110) > 0
         xx["dry"] = water == 0
         xx["wet"] = water == 128
+        xx = xx.drop_vars(["water"])
         return xx
 
     @staticmethod
     def _fuser(xx):
 
         wet = xx.wet
         dry = xx.dry
-        xx = _xr_fuse(xx.drop_vars(["wet", "dry"]), partial(_first_valid_np, nodata=NODATA), '')
-        xx["wet"] = _xr_fuse(wet, _fuse_or_np, wet.name)
-        xx["dry"] = _xr_fuse(dry, _fuse_and_np, wet.name)
+        bad = xx.bad
+
+        xx = _xr_fuse(xx.drop_vars(["wet", "dry", "bad"]), partial(_first_valid_np, nodata=NODATA), '')
+        
+        wet = _xr_fuse(wet, _fuse_or_np, wet.name)
+        dry = _xr_fuse(dry, _fuse_or_np, dry.name)
+        bad = _xr_fuse(bad, _fuse_or_np, dry.name)
+        
+        xx["wet"] = wet & (~dry) & (~bad)
+        xx["dry"] = dry & (~wet) & (~bad)
```

Comment:
I don't understand why these two lines are necessary. `wet`, `dry` and `bad` is mutually exclusive from the line `54-56`. Anything happened in line `67-71` makes it not the case?

Reply:
they are initially mutually exclusive for each individual input scene, but we fuse overlapping scenes from the same day with an `or` operation which which can cause them to not be mutually exclusive. E.g. say a pixel is in the overlap between two scenes and in one scene its dry and in the other its wet, then the fused pixel will be both wet and dry unless we do something like this